### PR TITLE
Feature: Enum type

### PR DESCRIPTION
**Pull request by basherru**
+ SmartCore::Types::Variadic::Enum

### CODE REVIEW

File: lib/smart_core/types/variadic.rb [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 # @api public
 # @since 0.3.0
+# @version 0.4.0
```
Comment (0exp) [inline lib/smart_core/types/variadic.rb:5]: this comment is addressed to Variadic constant only. but Variadic constant has no changes in current PR. so, this comment is unnecessary.

File: spec/types/variadic/enum_spec.rb [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        expect { type.validate!(:test) }.to raise_error(SmartCore::Types::TypeError)
+        expect { type.validate!(18483.991238) }.to raise_error(SmartCore::Types::TypeError)
+        expect { type.validate!(99999999999) }.to raise_error(SmartCore::Types::TypeError)
+        expect { type.validate!({}) }.to raise_error(SmartCore::Types::TypeError)
+        expect { type.validate!([]) }.to raise_error(SmartCore::Types::TypeError)
+        expect { type.validate!(Object.new) }.to raise_error(SmartCore::Types::TypeError)
+      end
+    end
+  end
+end
```
Comment (0exp) [inline spec/types/variadic/enum_spec.rb:83]: need to add examples with `BasicObject` instances

Comment (basherru) [reply spec/types/variadic/enum_spec.rb:87]: done

File: lib/smart_core/types/variadic.rb (hunk 1 above)
Comment (basherru) [reply lib/smart_core/types/variadic.rb]: ur right, done

File: README.md [hunk 3]
```diff
@@ -121,12 +121,14 @@ SmartCore::Types::Protocol::InstanceOf(::Time, ::DateTime, ::Date) # time or dat
 
 ```ruby
 SmartCore::Types::Variadic::Tuple
+SmartCore::Types::Variadic::Enum
```
Comment (0exp) [inline README.md:124]: Need to notify about new functionality in `Changelog.md` :)

Comment (0exp) [reply README.md:124]: My last comment here :)

File: lib/smart_core/types/variadic/enum.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+using SmartCore::Ext::BasicObjectAsObject
+
+# @api public
+# @since 0.4.0
```
Comment (0exp) [inline lib/smart_core/types/variadic/enum.rb:6]: 0.5.0 :)

**Comment by 0exp** [comment]
💯 👍

**Pull request merged**